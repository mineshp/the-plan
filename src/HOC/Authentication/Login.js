import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { loginUser } from '../../actions/authentication';
import { addNotification } from '../../actions/notification';
import LoginForm from '../../components/Authentication/Login';
import Auth from './Auth';

class Login extends Component {
    constructor(props, context) {
        super(props, context);

        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
        this.auth = new Auth();

        this.state = {
            username: null,
            password: null
        };
    }

    async loginUser() {
        const loginDetails = {
            username: this.state.username,
            password: this.state.password
        };

        const loginStatus = await this.props.actions.loginUser(loginDetails);

        this.props.actions.addNotification(this.props.notification);
        if (loginStatus.type === 'SUCCESS_LOGIN') {
            this.auth.setToken(loginStatus.token);
            this.context.router.history.push('/');
        } else {
            this.context.router.history.push('/user/login');
        }
    }

    handleChange(event) {
        const name = event.target.name;
        const value = event.target.value;
        this.setState({
            [name]: value
        });
    }

    async handleSubmit(event) {
        event.preventDefault();
        if (this.state.username && this.state.password) {
            return this.loginUser();
        }
        return undefined;
    }

    render() {
        return (
            <div>
                <LoginForm
                    handleChange={this.handleChange}
                    handleSubmit={this.handleSubmit}
                    data={this.state}
                />
            </div>
        );
    }
}

Login.propTypes = {
    actions: PropTypes.shape({
        loginUser: PropTypes.func.isRequired,
        addNotification: PropTypes.func.isRequired
    }),
    notification: PropTypes.shape({
        message: PropTypes.string,
        level: PropTypes.string,
        title: PropTypes.string
    })
};

Login.defaultProps = {
    actions: null,
    authentication: null,
    notification: null
};

Login.contextTypes = {
    router: PropTypes.object
};

/* istanbul ignore next: not testing mapStateToProps */
const mapStateToProps = (state) => (
    {
        authentication: state.authentication,
        notification: state.authentication.notification
    }
);

/* istanbul ignore next: not testing mapDispatchToProps */
const mapDispatchToProps = (dispatch) => (
    {
        actions: bindActionCreators({
            loginUser, addNotification
        }, dispatch)
    }
);

const LoginConnectedComponent = connect(
    mapStateToProps,
    mapDispatchToProps
)(Login);

export {
    Login,
    LoginConnectedComponent
};
