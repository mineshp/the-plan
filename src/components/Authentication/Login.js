import React from 'react';
import PropTypes from 'prop-types';
import { Button, Container, Form, Grid, Header, Image, Input, Message, Segment } from 'semantic-ui-react';
import logo from '../../md-logo-green.png';

const Login = ({
    handleSubmit,
    handleChange,
    data
}) => (
    <Container>
        <Grid stackable centered>
            <Grid.Column className="login-form-grid-column">
                <Header as="h2" color="teal" textAlign="center">
                    <Image src={logo} className="home-logo" />
                    {' '}Login to morpheus
                </Header>
                <Form size="large" onSubmit={handleSubmit}>
                    <Segment stacked>
                        <Form.Field>
                            <Input
                                fluid
                                name="username"
                                autoComplete="username"
                                placeholder="Username..."
                                icon="user"
                                iconPosition="left"
                                className="text-box-3-col"
                                defaultValue={data.username}
                                onChange={handleChange}
                            />
                        </Form.Field>
                        <Form.Field>
                            <Input
                                fluid
                                icon="lock"
                                iconPosition="left"
                                name="password"
                                type="password"
                                autoComplete="password"
                                placeholder="Password..."
                                className="text-box-3-col"
                                onChange={handleChange}
                                defaultValue={data.password}
                            />
                        </Form.Field>
                        <Button color="pink" fluid size="large" type="submit">Login</Button>
                    </Segment>
                </Form>
                <Message>
                New user? <a href="/user/register">Register</a>
                </Message>
            </Grid.Column>
        </Grid>
    </Container>
);

Login.propTypes = {
    handleSubmit: PropTypes.func.isRequired,
    handleChange: PropTypes.func.isRequired,
    data: PropTypes.shape({}).isRequired
};

export default Login;
