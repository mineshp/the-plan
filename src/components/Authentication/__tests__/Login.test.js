import React from 'react';
import { shallow } from 'enzyme';
import Login from '../Login';

describe('Login component', () => {
    it('renders correctly with valid data', () => {
        const handleChange = jest.fn();
        const handleSubmit = jest.fn();

        const data = {
            username: '',
            password: ''
        };

        const LoginComponent = shallow(
            <Login
                handleChange={handleChange}
                handleSubmit={handleSubmit}
                data={data}
            />
        );

        expect(LoginComponent).toMatchSnapshot();
    });
});
