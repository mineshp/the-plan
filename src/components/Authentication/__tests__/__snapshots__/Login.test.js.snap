// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component renders correctly with valid data 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Container
    className="content-body"
>
    <Grid
        centered={true}
        stackable={true}
    >
        <GridColumn
            className="login-form-grid-column"
        >
            <Header
                as="h2"
                className="login-header"
                textAlign="center"
            >
                <Image
                    as="img"
                    className="home-logo"
                    src="md-logo-green.png"
                    ui={true}
                />
                 
                Login to morpheus
            </Header>
            <Form
                as="form"
                onSubmit={[Function]}
                size="large"
            >
                <Segment
                    stacked={true}
                >
                    <FormField>
                        <Input
                            autoComplete="username"
                            className="text-box-3-col"
                            defaultValue=""
                            fluid={true}
                            icon="user"
                            iconPosition="left"
                            name="username"
                            onChange={[Function]}
                            placeholder="Username..."
                            type="text"
                        />
                    </FormField>
                    <FormField>
                        <Input
                            autoComplete="password"
                            className="text-box-3-col"
                            defaultValue=""
                            fluid={true}
                            icon="lock"
                            iconPosition="left"
                            name="password"
                            onChange={[Function]}
                            placeholder="Password..."
                            type="password"
                        />
                    </FormField>
                    <Button
                        as="button"
                        color="pink"
                        fluid={true}
                        size="large"
                        type="submit"
                    >
                        Login
                    </Button>
                </Segment>
            </Form>
            <Message>
                New user? 
                <Link
                    replace={false}
                    to="/user/register"
                >
                    Register
                </Link>
            </Message>
        </GridColumn>
    </Grid>
</Container>,
  "nodes": Array [
    <Container
      className="content-body"
>
      <Grid
            centered={true}
            stackable={true}
      >
            <GridColumn
                  className="login-form-grid-column"
            >
                  <Header
                        as="h2"
                        className="login-header"
                        textAlign="center"
                  >
                        <Image
                              as="img"
                              className="home-logo"
                              src="md-logo-green.png"
                              ui={true}
                        />
                         
                        Login to morpheus
                  </Header>
                  <Form
                        as="form"
                        onSubmit={[Function]}
                        size="large"
                  >
                        <Segment
                              stacked={true}
                        >
                              <FormField>
                                    <Input
                                          autoComplete="username"
                                          className="text-box-3-col"
                                          defaultValue=""
                                          fluid={true}
                                          icon="user"
                                          iconPosition="left"
                                          name="username"
                                          onChange={[Function]}
                                          placeholder="Username..."
                                          type="text"
                                    />
                              </FormField>
                              <FormField>
                                    <Input
                                          autoComplete="password"
                                          className="text-box-3-col"
                                          defaultValue=""
                                          fluid={true}
                                          icon="lock"
                                          iconPosition="left"
                                          name="password"
                                          onChange={[Function]}
                                          placeholder="Password..."
                                          type="password"
                                    />
                              </FormField>
                              <Button
                                    as="button"
                                    color="pink"
                                    fluid={true}
                                    size="large"
                                    type="submit"
                              >
                                    Login
                              </Button>
                        </Segment>
                  </Form>
                  <Message>
                        New user? 
                        <Link
                              replace={false}
                              to="/user/register"
                        >
                              Register
                        </Link>
                  </Message>
            </GridColumn>
      </Grid>
</Container>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Login
        data={
                Object {
                        "password": "",
                        "username": "",
                      }
        }
        handleChange={[Function]}
        handleSubmit={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "data": Object {
            "password": "",
            "username": "",
          },
          "handleChange": [Function],
          "handleSubmit": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Container
          className="content-body"
>
          <Grid
                    centered={true}
                    stackable={true}
          >
                    <GridColumn
                              className="login-form-grid-column"
                    >
                              <Header
                                        as="h2"
                                        className="login-header"
                                        textAlign="center"
                              >
                                        <Image
                                                  as="img"
                                                  className="home-logo"
                                                  src="md-logo-green.png"
                                                  ui={true}
                                        />
                                         
                                        Login to morpheus
                              </Header>
                              <Form
                                        as="form"
                                        onSubmit={[Function]}
                                        size="large"
                              >
                                        <Segment
                                                  stacked={true}
                                        >
                                                  <FormField>
                                                            <Input
                                                                      autoComplete="username"
                                                                      className="text-box-3-col"
                                                                      defaultValue=""
                                                                      fluid={true}
                                                                      icon="user"
                                                                      iconPosition="left"
                                                                      name="username"
                                                                      onChange={[Function]}
                                                                      placeholder="Username..."
                                                                      type="text"
                                                            />
                                                  </FormField>
                                                  <FormField>
                                                            <Input
                                                                      autoComplete="password"
                                                                      className="text-box-3-col"
                                                                      defaultValue=""
                                                                      fluid={true}
                                                                      icon="lock"
                                                                      iconPosition="left"
                                                                      name="password"
                                                                      onChange={[Function]}
                                                                      placeholder="Password..."
                                                                      type="password"
                                                            />
                                                  </FormField>
                                                  <Button
                                                            as="button"
                                                            color="pink"
                                                            fluid={true}
                                                            size="large"
                                                            type="submit"
                                                  >
                                                            Login
                                                  </Button>
                                        </Segment>
                              </Form>
                              <Message>
                                        New user? 
                                        <Link
                                                  replace={false}
                                                  to="/user/register"
                                        >
                                                  Register
                                        </Link>
                              </Message>
                    </GridColumn>
          </Grid>
</Container>,
        "_debugID": 2,
        "_renderedOutput": <Container
          className="content-body"
>
          <Grid
                    centered={true}
                    stackable={true}
          >
                    <GridColumn
                              className="login-form-grid-column"
                    >
                              <Header
                                        as="h2"
                                        className="login-header"
                                        textAlign="center"
                              >
                                        <Image
                                                  as="img"
                                                  className="home-logo"
                                                  src="md-logo-green.png"
                                                  ui={true}
                                        />
                                         
                                        Login to morpheus
                              </Header>
                              <Form
                                        as="form"
                                        onSubmit={[Function]}
                                        size="large"
                              >
                                        <Segment
                                                  stacked={true}
                                        >
                                                  <FormField>
                                                            <Input
                                                                      autoComplete="username"
                                                                      className="text-box-3-col"
                                                                      defaultValue=""
                                                                      fluid={true}
                                                                      icon="user"
                                                                      iconPosition="left"
                                                                      name="username"
                                                                      onChange={[Function]}
                                                                      placeholder="Username..."
                                                                      type="text"
                                                            />
                                                  </FormField>
                                                  <FormField>
                                                            <Input
                                                                      autoComplete="password"
                                                                      className="text-box-3-col"
                                                                      defaultValue=""
                                                                      fluid={true}
                                                                      icon="lock"
                                                                      iconPosition="left"
                                                                      name="password"
                                                                      onChange={[Function]}
                                                                      placeholder="Password..."
                                                                      type="password"
                                                            />
                                                  </FormField>
                                                  <Button
                                                            as="button"
                                                            color="pink"
                                                            fluid={true}
                                                            size="large"
                                                            type="submit"
                                                  >
                                                            Login
                                                  </Button>
                                        </Segment>
                              </Form>
                              <Message>
                                        New user? 
                                        <Link
                                                  replace={false}
                                                  to="/user/register"
                                        >
                                                  Register
                                        </Link>
                              </Message>
                    </GridColumn>
          </Grid>
</Container>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Login
    data={
        Object {
            "password": "",
            "username": "",
          }
    }
    handleChange={[Function]}
    handleSubmit={[Function]}
/>,
}
`;
